Include/Rover/Range.hpp

```diff
@@ -132,9 +136,13 @@ namespace Rover {
     auto begin = evaluator.evaluate(m_begin);
     auto end = evaluator.evaluate(m_end);
     while(true) {
-      auto random_fraction = m_distribution(m_engine);
-      auto distance = static_cast<Type>((end - begin) * random_fraction);
-      auto value = begin + distance;
+      auto value = calculate_random(begin, end);
```

Comment:
The logic is a bit indirect so let's take an example. Range(0, 10, CLOSED) has to produce each value from 0, 1, ..., 10 with a 1/11 chance. When we use a real distribution from [0.0, 1.0) and scaling it to 0-10, we instead generate values from 0, 1, ..., 9 with a probability of 1/10 each (that generation is now done by the calculate_random function).
The idea is to reserve 0 to represent both begin and end, and generate it with a 2/11 chance - after that, we split that chance between begin and end as 50:50 reaching the 1/11 for begin and end, and, as long as the rest values are equally likely, they will have a 1/11 probability each as well. To achieve that 2/11 chance, I pick a random value from [0, 10) twice. Then, I analyse the joint probabilities of both values. By excluding the case of val1 = 0, val2 = 0 and treating any other val1 = 0 || val2 = 0 as a zero-case, I get the following probability for the zero-case: prob = (9 + 9) / (10^2 - 1) = 18/99 = 2/11 as required. I have to drop the val1 = 0, val2 = 0 case to decrement the denominator.